cogs/archive.py

```diff
@@ -176,17 +177,38 @@ async def archive_category(
 
             await channel.edit(sync_permissions=True)
 
-        overwrite: discord.Member | discord.Role
-        for overwrite in category.overwrites:
-            await category.set_permissions(overwrite, overwrite=None)
+        target: discord.Member | discord.Role
+        for target in category.overwrites:
+            await category.set_permissions(target=target, overwrite=None)
+
+        try:
+            everyone_role: discord.Role = await ctx.bot.get_everyone_role()
+            await category.set_permissions(
+                target=everyone_role,
+                view_channel=False,
+            )
+        except EveryoneRoleCouldNotBeRetrievedError:
+            logger.exception(
+                "Could not retrieve the @everyone role when archiving category %r",
+                category.name,
+            )
```

Comment:
`try` blocks should only contain the code to have exception handling, use `else` block for the "happy path".

```suggestion
        try:
            everyone_role: discord.Role = await ctx.bot.get_everyone_role()
        except EveryoneRoleCouldNotBeRetrievedError:
            logger.exception(
                "Could not retrieve the @everyone role when archiving category %r",
                category.name,
            )
        else:
            await category.set_permissions(
                target=everyone_role, view_channel=False
            )
```